pkg/k8s/client.go

```diff
@@ -124,3 +124,15 @@ func (a *Access) DeleteObject(ctx context.Context, object, owner metav1.Object)
 
 	return nil
 }
+
+// IsJobFinished returns true if the given job has already succeeded or failed
+func (a *Access) IsJobFinished(namespacedName types.NamespacedName) (finished bool, err error) {
+	job, err := a.Clientset.BatchV1().Jobs(namespacedName.Namespace).Get(
+		namespacedName.Name, metav1.GetOptions{})
+	if err != nil {
+		return false, err
+	}
+
+	finished = job.Status.Succeeded+job.Status.Failed > 0
```

Comment:
This might not be correct if the job is re-tried:
Both Succeeded and Failed are integers, counting how many of the jobs finished in the given state.
In case of re-tried job it will return true, while the job is still running.

Reply:
That's normally true, but in case of fio retry is disabled. What's your suggestion here? Close this PR and leave isJobFinished() as was, fio-specific?

Reply:
given that we're moving to a generic package I'd not rely on the fact that within fio is correct.

I think we could do two things:
1) check for Status.completionTime
2) do some math: from completions, backofflimit, succeeded & failed figure out if it will not be scheduled anymore

I'd opt for the first approach, based on this thread:
https://github.com/kubernetes/kubernetes/issues/68712

Reply:
Went with 1. :)